operator/executor_start.sh

```diff
@@ -0,0 +1,224 @@
+#! /bin/bash
+#
+# start.sh
+# shellcheck disable=SC1090
+
+PLAYBOOK_CONFIG_PATH=${PLAYBOOK_CONFIG_PATH:-/opt/config}
+PLAYBOOK_CONFIG_FILE=${PLAYBOOK_CONFIG_FILE:-config.sh}
+
+RETRIES=${RETRIES:-3}
+RETRIES_INTERVAL=${RETRIES_INTERVAL:-10}
+
+SSH_USER_NAME=${SSH_USER_NAME:-ansible}
+SSH_KEYS_PATH=${SSH_KEYS_PATH:-keys}
+
+
+function logs_format(){
+    level="${1}"
+    shift 
+    msg="${*}"
+    echo "{\"time\": \"$(date -Iseconds)\", \"level\": \"${level}\", \"msg\": \"${msg}\", \"namespace\": \"${ANSIBLE_EXECUTOR_NAMESPACE}\", \"name\": \"${ANSIBLE_EXECUTOR_NAME}\"}" >&2
+}
+
+
+function ssh_config(){
+    cat << EOF >> "/home/${SSH_USER_NAME}/.ssh/config"
+Host *
+    StrictHostKeyChecking no
+    UserKnownHostsFile /dev/null
+EOF
+    
+    for key in /home/"${SSH_USER_NAME}"/"${SSH_KEYS_PATH}"/*; do
+        logs_format info "Configuring SSH client with key ${key}"
+        if [ -d "${key}/config" ]; then
+            logs_format info "Adding SSH client configuration provided in configmap using identity file ${key}/ssh-privatekey"
+            cat << EOF >> "/home/${SSH_USER_NAME}/.ssh/config"
+$(cat "${key}/config/ssh_config")
+  IdentityFile ${key}/ssh-privatekey
+EOF
+        fi
+        if  [ -d "${key}/secret-config" ]; then
+            logs_format info "Adding SSH client configuration provided in secret using identity file ${key}/ssh-privatekey"
+            cat << EOF >> "/home/${SSH_USER_NAME}/.ssh/config"
+$(cat "${key}/secret-config/ssh_config")
+  IdentityFile ${key}/ssh-privatekey
+EOF
+        fi
+        if [ ! -d "${key}/config" ] && [ ! -d "${key}/secret-config" ]; then
+            logs_format info "No configuration provided, identity file ${key}/ssh-privatekey will used for all hosts"
+            cat << EOF >> "/home/${SSH_USER_NAME}/.ssh/config"
+Host *
+  IdentityFile ${key}/ssh-privatekey
+EOF
+        fi
+    done
+
+    logs_format info "SSH client configured, setting restricted permissions for file /home/${SSH_USER_NAME}/.ssh/config"
+    chmod 0600 "/home/${SSH_USER_NAME}/.ssh/config"
+}
+
+function git_repo_setup(){
+    logs_format info "Attempting to clone git repository ${GIT_REPO} with ${RETRIES} retries"
+    for retry in $(seq 1 "${RETRIES}"); do
+        if git clone "${GIT_REPO}" "${PLAYBOOK_CONFIG_PATH}/repo" &>/dev/null;
```

Comment:
Make it so we are cloning only a specific commit (e.g. specified tag or latest commit in a branch).